Ubuntu 3.2 wip

mihgen opened the pull request:
Ongoing work on Ubuntu support. Let's review, test that CentOS is not broken by this, and merge.

demofly committed:
FUEL-960: Mirantis repos to support OpenStack Grizzly on Ubuntu

xdel committed:
Merge pull request #257 from demofly/ubuntu-3.1

FUEL-960: Mirantis repos to support OpenStack Grizzly on Ubuntu

demofly committed:
FUEL-748: switching to a new quantum-ovs-cleanup style - it is embedded into quantum-plugin-openvswitch-agent upstart script

xdel committed:
Merge pull request #258 from demofly/ubuntu-3.1

FUEL-748: switching to a new quantum-ovs-cleanup style - it is embedded ...

xdel committed:
repo to the standard layout

xdel committed:
no writables in /usr

xdel committed:
Merge pull request #260 from xdel/remove-unsecure-chown

no writables in /usr

xdel committed:
Merge with master repo

xdel committed:
No more warnings on the puppetlabs repo

xdel committed:
Merge pull request #267 from xdel/ubuntu-3.1

Merge with master, cosmetic additions

xdel committed:
logdir for horizon

xdel committed:
Merge pull request #268 from xdel/ubuntu-3.1

logdir for horizon

xdel committed:
increased default syslog msg size, FUEL-742

xdel committed:
Merge branch 'ubuntu-3.1' of github.com:Mirantis/fuel into ubuntu-3.1

Commit:
Add timeout for deployed nodes

xdel committed:
l3 workaround from xenolog/fuel

xdel committed:
Merge pull request #281 from xdel/ubuntu-3.1

l3 workaround from xenolog/fuel

xdel committed:
Those services should not be started for sure

xdel committed:
metadata-agent noautostart on Debian

xdel committed:
Revert on boot override for quantum-server

xdel committed:
Merge pull request #291 from xdel/ubuntu-3.1

Revert on boot override for quantum-server

dmi-try committed:
Remove old versions of cobbler manifests for rhel

xdel committed:
merge ubuntu-3.1 with real 3.1

xdel committed:
missed on merge

xdel committed:
missed on merge

xdel committed:
missed on merge

xdel committed:
bad declaration

xdel committed:
ubuntu-related fixes

xdel committed:
Merge branch 'fix-service-declaration' into ubuntu-3.1-real

xdel committed:
very crappy fix

xdel committed:
fix for horizon

xdel committed:
shebang for rebuild function in mysql-wss

xdel committed:
wsrep fix

xdel committed:
only RH derivatives has a seperate init

xdel committed:
missing anchor

andrei4ka committed:
switching to new repository

xdel committed:
Merge pull request #2 from andrei4ka/ubuntu-3.1-real-andreika

Ubuntu 3.1 real andreika

andrei4ka committed:
remove koji centos repo

xdel committed:
Merge pull request #3 from andrei4ka/ubuntu-3.1-real-andreika

Ubuntu 3.1 real andreika

Commit:
Merge pull request #489 from aglarendil/master

FF do develop branch

xdel committed:
merge with 3.1

andrei4ka committed:
corosync+haproxy fix

corosync+haproxy fix

xdel committed:
Merge pull request #6 from andrei4ka/patch-1

corosync+haproxy fix

Commit:
renamed variables
    mco_stomphost => $mco_host
    mco_stompport => $mco_port
    mco_stompuser => $mco_user
    mco_stomppassword => $mco_password

variable mco_port was predifined because cobbler know nothing about it

xdel committed:
Merge pull request #7 from gleb108/ubuntu-3.1-real

fix kickstart

xdel committed:
Merge pull request #494 from xdel/ubuntu-3.1-real

move stuff to the mirantis repo

xdel committed:
partial revert of 5cea30ba6cd538be235bd61d8e71f923400c2900

xdel committed:
partial revert of 4b69a348

aglarendil commented on deployment/puppet/cobbler/templates/preseed/ubuntu-1204.preseed.erb:
```diff
@@ -181,4 +181,7 @@ $SNIPPET('ubuntu_puppet_config') && \
 $SNIPPET('ubuntu_mcollective_config') && \
 echo -e "nf_conntrack_ipv4\nnf_conntrack_ipv6" >> /target/etc/modules && \
 echo -e "net.nf_conntrack_max=1048576" >> /target/etc/sysctl.conf && \
+in-target wget http://apt.puppetlabs.com/pubkey.gpg && \
```
instead do wget -O - <URL> | apt-key add -

aglarendil commented on deployment/puppet/quantum/manifests/agents/l3.pp:
```diff
@@ -34,6 +34,18 @@
+  {
+    'Debian':
+      {
+       file { "/etc/init/quantum-l3-agent.override":
```
why should not you use enable=> "false" upstart services ?

xdel replied:
Unfortunately upstart brings them up by itself then at least once.

aglarendil commented on deployment/puppet/quantum/manifests/agents/ovs.pp:
```diff
@@ -29,6 +29,19 @@
+  {
+    'Debian':
+      {
+       file { "/etc/init/quantum-plugin-openvswitch-agent.override":
```
why should not you use enable=> "false" upstart services ?

aglarendil replied:
WTF? You should use operatingsystem == 'Ubuntu' instead of Debian.

aglarendil commented on deployment/puppet/quantum/manifests/server.pp:
```diff
@@ -18,6 +18,18 @@
+  {
+    'Debian':
+      {
+       file { "/etc/init/quantum-metadata-agent.override":
```
why should not you use enable=> "false" upstart services ?

aglarendil replied:
WTF? You should use operatingsystem == 'Ubuntu' instead of Debian.

aglarendil commented on deployment/puppet/quantum/manifests/agents/dhcp.pp:
```diff
@@ -19,6 +19,19 @@
 
+  case $::osfamily
+  {
+    'Debian':
```
WTF? You should use operatingsystem == 'Ubuntu' instead of Debian.

xdel replied:
IMO does not worth it. Most of our OS selector cases will produce broken stuff anyway deploying on unsupported distros like Fedora.

aglarendil replied:
There is no reason to continue such practice.

xenolog commented on deployment/puppet/cluster/manifests/haproxy.pp:
```diff
@@ -62,11 +62,10 @@
 
   if ($::osfamily == 'Debian') {
     Package['haproxy'] ->
-      Service['haproxy-init-stopped'] ->
```
Why? 
system service MUST be stopped. ALWAYS!!!

ha-proxy managed by pacemaker, not by system init scripts.

xdel replied:
it is managed already by string right below

xenolog commented on deployment/puppet/quantum/manifests/agents/ovs.pp:
```diff
@@ -205,6 +219,15 @@
+    }
+    Service['quantum-plugin-ovs-service'] -> Service['quantum-ovs-agent-cleanup']
+  } else {
+    anchor{'quantum-ovs-agent-done': }
```
Why you hide anchor under IF ???

xdel replied:
it should be marked as 'toberewritten'. There was unconditional dependency on this anchor way hard to ifify.

andrei4ka committed:
Directory dependence fix

aglarendil committed:
[PRD-1499] Assign keystone_ec2_url value

Commit:
Changed repository for puppet packages
(packages moved from puppetlabs to mirantis)
fixed version for the puppetlabs' packages

andrei4ka committed:
Keystone provider fix PRD-1524

aglarendil committed:
remove obsolete debug in firewall module

Commit:
Merge pull request #501 from Mirantis/prd1302

remove obsolete debug in firewall module

andrei4ka committed:
tinyproxy fix for ubuntu

Commit:
Merge pull request #498 from Mirantis/prd1499

[PRD-1499] Assign keystone_ec2_url value

andrei4ka committed:
cirros-testvm fix for ubuntu

xdel committed:
Merge pull request #500 from gleb108/ubuntu-3.2-WIP

Changed repository for puppet packages; this is WIP commit - repo should be pointed to master later.

Commit:
add ruby gems, vlan, nailgun.* packages

xdel committed:
Merge pull request #502 from gleb108/ubuntu-3.2-WIP

add ruby gems, vlan, nailgun.* packages

andrei4ka committed:
some fixes

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel-syntax-check/2/

xdel committed:
Merge remote-tracking branch 'origin/master' into ubuntu-3.2-WIP

xdel committed:
merge with develop

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel-syntax-check/12/

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel-syntax-check/13/

r0mik committed:
add rubygem json for rhel

r0mik committed:
add rubygem json for rhel

Commit:
fix deploy

andrei4ka committed:
corosync remove override fix

andrei4ka committed:
corosync force restart before galera

andrei4ka committed:
chsnged dependancies for exec

Commit:
add necessary patched file to ceph-deploy

added files/ceph-deploy
added file for /usr/bin/ceph-deploy
cleaned some trailing whitespace

Commit:
fix ceph-deploy file; fix keystone class

Commit:
Added support for disk labels and unmounted partitions

Commit:
fix ceph-deploy source; fix deploy push config

Commit:
Determine list of OSD devices dynamically

xdel committed:
Merge pull request #507 from r0mik/rhos_json

Add  rubygem-json to RHOS repos

Commit:
expanded readme; fix race in deps.pp

extended README with cinder examples
removed some todo's from site.pp
hard coded file require

angdraug committed:
not reinstall ntp in site.pp so that cobbler's ntp.conf changes are not overwritten

angdraug committed:
typo in ceph README: rdb -> rbd

Commit:
fix nova_compute

Commit:
fix readme whitespace for the ...

Commit:
replace tinyproxy to apache

mihgen committed:
Merge remote branch 'origin/develop'

xdel committed:
those files should be licensed for sure

xdel committed:
Merge pull request #515 from xdel/licenses

those files should be licensed for sure

dmi-try committed:
Merge remote-tracking branch 'origin/master' into no-rhel-ks

Conflicts:
	deployment/puppet/cobbler/manifests/distro/rhel63-x86_64.pp
	deployment/puppet/cobbler/manifests/distro/rhel63_x86_64.pp
	deployment/puppet/cobbler/manifests/profile/rhel63-x86_64.pp
	deployment/puppet/cobbler/manifests/profile/rhel63_x86_64.pp
	deployment/puppet/cobbler/templates/kickstart/rhel.ks.erb

xdel committed:
Merge pull request #514 from Zipfer/apache_api_proxy

replace tinyproxy to apache

Commit:
add synax check

Commit:
Get rid of set -e, set +e, etc. in syntax check script

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/10/

Commit:
Fix Anaconda issues with leftover LVM metadata prior to installation

Commit:
persisitent reconnect to database was added

Commit:
add options report=false and configtimeout = 600 to puppet.conf

Commit:
Openstack now doploing successfully ;)

xdel committed:
Merge pull request #520 from gleb108/ubuntu-3.2-WIP

persistent reconnect to database and puppet fixes

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/22/

xdel committed:
Merge pull request #522 from Mirantis/develop

Update master for #492

xdel committed:
remove call to deleted class

xdel committed:
Merge pull request #525 from xdel/untiny

remove call to deleted class

Commit:
Fixed typo in LVM metadata commands

Commit:
fix syntax mysql replicator

Commit:
fix syntax ceph deploy.pp

Commit:
Merge branch 'ubuntu-3.2-WIP' of github.com:Mirantis/fuel into Mirantis-ubuntu-3.2-WIP

Conflicts:
	deployment/puppet/osnailyfacter/manifests/tinyproxy.pp

Commit:
Merge pull request #518 from Zipfer/fuel_syntax_check

fuel synax check

sirmax123 committed:
added heat module

sirmax123 committed:
added savanna module

sirmax123 committed:
added murano module

nikit0z committed:
added mysql connection retries

Commit:
removed debug messages

Commit:
added jenkins syntax checker

Commit:
o-o-ops...  =)

Commit:
unicast false by default

Commit:
default script

Commit:
fix

Commit:
partial sync with master

Commit:
partial sync with master

Commit:
haproxy ocf file fix for ubuntu

Commit:
haproxy ocf file fix for ubuntu

Commit:
fix on fix on fix

Commit:
fix on fix on fix on fix

Commit:
fix on fix on fix on fix on fix

xdel committed:
Merge remote-tracking branch 'origin/master' into ubuntu-3.2-WIP

nikit0z committed:
Merge remote-tracking branch 'upstream/ubuntu-3.2-WIP' into ubuntu-3.2-WIP

Commit:
new class for clonecleaanup

Commit:
Merge branch 'ubuntu-3.2-WIP' of github.com:andrei4ka/fuel into ubuntu-3.2-WIP

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/54/

xdel committed:
partial revert of a5932a579ff941f48e1e59c0b8df0861ef7cb7c9

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/61/

xdel committed:
Merge pull request #526 from sirmax123/heat

added heat module

xdel committed:
Merge pull request #528 from sirmax123/savanna

added savanna module

xdel committed:
Merge pull request #527 from sirmax123/murano

added murano module

xdel committed:
Merge pull request #530 from nikit0z/ubuntu-3.2-WIP

added mysql connection retries

xdel committed:
Merge pull request #516 from dmi-try/no-rhel-ks

Remove old versions of cobbler manifests for rhel

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/62/

angdraug committed:
fixed typos in comments

angdraug committed:
use "cephosd" label for osd to differentiate from osd journal

xdel committed:
Merge pull request #531 from xarses/ceph

Sync ceph working branch

xdel committed:
couple of reverts

xdel committed:
merge with ubuntu-3.2-wip

xdel committed:
Merge pull request #497 from andrei4ka/ubuntu-3.2-WIP

Directory dependence fix

xdel committed:
upstart-only workarounds

xdel committed:
dhcp agent boot override

xdel committed:
Merge branch 'master' into ubuntu-3.2-WIP

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/71/

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/72/

xdel committed:
keystone dependency fix

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/73/
